Add text clarifying test executable permissions (#2879)
exercism/java: docs/TESTS.md

   _(Don't worry about the tests failing, at first, this is how you begin each exercise.)_

   If you get the following error:

>    ./gradlew: Permission denied

Reviewer: Could we put this in a code block (using the triple back ticks ```), like the earlier snippets?

Author: Sure! Let me just rebase this.

Author: Ah, the previous ones had blocks, I didn't see it because I have markup turned off 😅.

   Then that means there's an issue regarding permissions, that the file isn't executable. To fix this, run:

>    chmod +x ./gradlew

Reviewer: Similar, could we put this in a code block?

   _(Don't worry about the tests failing, at first, this is how you begin each exercise.)_

   If you get the following error:

   ```sh
   ./gradlew: Permission denied
   ```

>    Then that means there's an issue regarding permissions, that the file isn't executable. To fix this, run:

Reviewer: I wanted to suggest a more direct wording for the first sentence. What do you think? 

```suggestion
   Then the file is missing the execute permission. To fix this, run:
```

Author: I agree, this wording seems better.
